gm4_metallurgy/beet.yaml

```diff
@@ -63,3 +67,108 @@ meta:
+        broadcast:
+          - reference: block/mould/cool_empty
+          - reference: block/mould/hot_empty
+          - reference: block/mould/cool_metal
+          - reference: block/mould/hot_metal
+      - item: obsidian
+        reference: item/slightly_damaged_obsidian
+        template: vanilla
+      - reference: shamir/defuse
+        item: armor
+        template: 
+          name: shamir
+          metal: aluminium
+          textures_path: item/shamir/aluminium
+      - reference: shamir/moneo
+        item: [tools, elytra, shears]
+        template: 
+          name: shamir
+          metal: aluminium
+          textures_path: item/shamir/aluminium
+      - reference: shamir/arborenda
+        item: axes
+        template: 
+          name: shamir
+          metal: barium
+          textures_path: item/shamir/barium
+      - reference: shamir/sensus
+        item: compass
+        template: 
+          name: shamir
+          metal: barium
+          textures_path: item/shamir/barium
+      - reference: shamir/gemini
+        item: hoes
+        template: 
+          name: shamir
+          metal: barimium
+          textures_path: item/shamir/barimium
+      - reference: shamir/musical
+        item: [swords, trident]
+        template: 
+          name: shamir
+          metal: barimium
+          textures_path: item/shamir/barimium
+      - reference: shamir/forterra
+        item: pickaxes
+        template: 
+          name: shamir
+          metal: bismuth
+          textures_path: item/shamir/bismuth
+      - reference: shamir/hypexperia
+        item: [swords, pickaxes, axes, shovels, fishing_rod]
+        template: 
+          name: shamir
+          metal: bismuth
+          textures_path: item/shamir/bismuth
+      - reference: shamir/vibro
+        item: boots
+        template: 
+          name: shamir
+          metal: thorium
+          textures_path: item/shamir/thorium
+      - reference: shamir/ender_bolt
+        item: [swords, trident]
+        template: 
+          name: shamir
+          metal: thorium
+          textures_path: item/shamir/thorium
+      - reference: shamir/infinitas
+        item: buckets
+        template: 
+          name: shamir
+          metal: curies_bismium
+          textures_path: item/shamir/curies_bismium
+      - reference: shamir/tinker
+        item: [pickaxes, shovels, hoes, swords]
+        template: 
+          name: shamir
+          metal: curies_bismium
+          textures_path: item/shamir/curies_bismium
+      - item: lava_bucket
+        reference: gui/advancement/metallurgy_overheat_cast
+        template: advancement
+      - item: player_head
+        reference: gui/advancement/metallurgy_blast_ore
+        template:
+          name: advancement
+          forward: item/lump/bauxite
+      - item: player_head
+        reference: gui/advancement/metallurgy_cast
+        template:
+          name: advancement
+          forward: shamir/defuse/band
```

Comment:
I've been looking through this this morning and think we should re-think where we put these texture/asset definitions.

`beet.yaml` has so far been for meta-data, whereas this asset data isn't meta-data anymore if you think about it, it's resource definitions. That plus the fact that these make the `beet.yaml` extremely long and overwhelming to look at when a module introduces more than a couple of textures makes me think these should go in their own, separate `.yaml` file.

Reply:
Agreed. I think similar to how we have `translations.csv`, maybe `assets.yaml` or `assets/items.yaml` could work.

Reply:
So a couple of arguments for why I think it should stay in `beet.yaml`.
1. `beet.yaml` isn't for "module metadata", but rather "beet build metadata", with all the info required for beet plugins to work. 
2. Putting this data in `beet.yaml` lets its structure be easily read in and validated by beet. Malformed data structures get nice diagnostic data printed upon the beet build. Counter - I could reimplement this validation on my own.
3. The data here lets it be generated by a beet plugin easily, just by appending to the `ctx.meta['gm4.model_data']`. Most of the most "overwhelming" config is created this way, since those are also the most repetitive. Counter - I could create some file pinned to `ctx.data.extra` somewhere, but it feels strange to put it either on the datapack or texturepack, and it'd have to be cleaned up before we output.
4. Model data isn't just for defining items, but any CMD interactive assets, such as advancement icons or block models. About half our modules don't have any assets of their own, but do still define CMD for future compatibility or retexturing packs.

Reply:
re: end of 3, how would moving this model_data into a standalone file change whether it is put on either the datapack or resource pack, if the format stays the same? (since we just want to move this list into a separate file)

Reply:
Beet requires all files to be members of either the ResourcePack() or DataPack(), so any standalone file would need to belong to one of these to be accessed by plugins. `meta` is just a general purpose dict present on the `Context`, so its nicer to access by plugins.  

Its also just organizationally weird to have an asset that affects both the DP and RP only be a member of one of those containers, but I can get over that

Reply:
An update on this thread from today's meeting: I'm going to add a config file that gets merged into `meta`, that way modules with very long model_data can move it out into a separate file to not clutter the beet.yaml, but modules with one or two definitions can still keep it in the beet.yaml directly.